### ISSUE TITLE
Null deref in cmd_search.c:2031 

### ISSUE THREAD

**Issue by fcasal** [radare/radare2#10050]
### Work environment

| Questions                                            | Answers
|------------------------------------------------------|--------------------
| OS/arch/bits                             | 16.04.1 Ubuntu x86_64
| File format of the file you reverse      | ELF
| Architecture/bits of the file            | x86/64
| r2 -v          | radare2 2.6.0-git 18193 @ linux-x86-64 git.2.5.0-206-gbb2f7c9 commit: bb2f7c92117e5d9dbe0b6e83162411cae260cc16 build: 2018-05-08__11:49:23
| r2 -V | https://gist.github.com/fcasal/75b79f13ad691d3145cdd7b8b1c4b73a

### TLDR
Segfault caused by null deref in search

### Steps to reproduce the behaviour
```
mandlebro@machine$ r2 --
[0x00000000]> /c
Segmentation fault (core dumped)
```

The offending line is 2031 of cmd_search.c
```c
2029		if (!param->boundaries) {
2030			map = R_NEW0 (RIOMap);
2031			map->fd = core->io->desc->fd;
```
where core->io->desc is NULL.

**Comment by pawlos** [comment]
that was this check for that was removed in this one https://github.com/radare/radare2/issues/9992

**Comment by radare** [comment]
We should use the r_io_fd_get_current api instead of accessing those fields

**Comment by radare** [comment]
But assuming threre should be a desc pointer in io is also wrong. This field must die in siol2
